app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -89,6 +89,11 @@ def publish
         fail HttpErrors::BadRequest, _("Directly setting package lists on composite content views is not allowed. Please " \
                                      "update the components, then re-publish the composite.")
       end
+      if params[:major].present? && params[:minor].present? && ContentViewVersion.find_by(:content_view_id => params[:id], :major => params[:major], :minor => params[:minor]).present?
+        major = params[:major]
+        minor = params[:minor]
+        fail HttpErrors::BadRequest, _("A CV version already exists with the same major and minor version (#{major}.#{minor})")
```

Comment:
I would do just `#{params[:major]}.#{params[:minor]}` so you dont have to do the 2 lines above